### PR TITLE
PMM-12045 transition to unified grafana binary

### PR DESCRIPTION
**Pull request by atymchuk**
PMM-12045

Link to the Feature Build: SUBMODULES-3344

The code contains a number of grafana configuration overrides like `cfg:default.paths.plugins=/srv/grafana/plugins` and many others, which are set in at least three different places. It's difficult to reason about the source of truth, and it'll be hard to maintain the overrides going forward.

https://github.com/percona/pmm/pull/2368/files#diff-f86b15415827cbf6cd2001347ab4faa6ede35239f865571c0f0c71814414836aL10-L15

> __Warning__ 
Note: pls don't update the branch - it will fail `managed` tests in such case.

### CODE REVIEW

File: build/packages/deb/postinst [hunk 1]
```diff
@@ -25,7 +25,7 @@ case "$1" in
     ;;
 
     *)
-        echo "postinst called with unknown argument \`$1'" >&2
+        echo "postinst called with unknown argument '$1'" >&2
```
Comment (atymchuk) [inline build/packages/deb/postinst:28]: Corrected a typo

File: build/ansible/roles/pmm2-images/files/grafana.ini [hunk 2]
```diff
@@ -4,15 +4,9 @@
...
-    /usr/sbin/grafana-server
+    /usr/sbin/grafana server
         --homepath=/usr/share/grafana
         --config=/etc/grafana/grafana.ini
-        cfg:default.paths.data=/srv/grafana
-        cfg:default.paths.plugins=/srv/grafana/plugins
-        cfg:default.paths.logs=/srv/logs
-        cfg:default.log.mode=console
-        cfg:default.log.console.format=console
-        cfg:default.server.root_url="https://%%(domain)s/graph"
```
Comment (atymchuk) [inline build/ansible/roles/pmm2-images/files/grafana.ini:15]: These overrides should all be captured in one place - /etc/grafana/grafana.ini

File: update/ansible/playbook/tasks/roles/grafana/files/grafana.ini [hunk 3]
```diff
@@ -4,11 +4,29 @@
...
+[log.console]
+# log line format, valid options are text, console and json
+format = console
 
 #################################### Server ####################################
 [server]
 # enable gzip
 enable_gzip = true
+# The public facing domain name used to access grafana from a browser
+domain = 127.0.0.1
```
Comment (artemgavrilov) [inline update/ansible/playbook/tasks/roles/grafana/files/grafana.ini:27]: Will CLI arg overwrite this value? Cause we have domain setting on the UI.

**Review by artemgavrilov — APPROVED** [review]

File: update/ansible/playbook/tasks/roles/grafana/files/grafana.ini (hunk 3 above)
Comment (atymchuk) [reply update/ansible/playbook/tasks/roles/grafana/files/grafana.ini:27]: > Will CLI arg overwrite this value? Cause we have domain setting on the UI.

Short answer is yes.
Long answer: managed is basically generating a new `/etc/supervisord.d/grafana.ini` file, which launches `grafana server` with a number of what is called [configuration overrides](https://grafana.com/docs/grafana/latest/cli/#override-a-configuration-setting). The latter will take precedence over settings found in `/etc/grafana/grafana.ini`.

**Review by idoqo — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**